dace/codegen/compiled_sdfg.py

```diff
@@ -438,33 +473,93 @@ def _construct_args(self, kwargs) -> Tuple[Tuple[Any], Tuple[Any]]:
             argtypes = []
             argnames = []
             sig = []
+
         # Type checking
+        self._construct_args_type_checking(argnames, arglist, argtypes)
+
+        # Explicit casting
+        arg_ctypes = self._construct_args_explicit_casting(arglist, argtypes, kwargs)
+
+        # Creates the call parameters
+        callparams = self._construct_args_callparams(arglist, arg_ctypes, argtypes, argnames)
+
+        initargs = self._construct_args_initarg(callparams)
+
+        try:
+            # Replace arrays with their base host/device pointers
+            newargs = []
```

Comment:
Before I added `_fast_call()` I tried to improve `_construct_args()` I profiled it and the function as a whole was faster (but still too slow).
However, since I care more on `_fast_call()` I could revert these changes.

Reply:
Please do.

Reply:
See https://github.com/spcl/dace/pull/1467#discussion_r1436812693 for reason why not.